ratpack-dropwizard-metrics/src/main/java/ratpack/dropwizard/metrics/internal/TimedMethodInterceptor.java

```diff
@@ -38,12 +39,19 @@
   public Object invoke(MethodInvocation invocation) throws Throwable {
     String timerTag = buildTimerTag(invocation.getMethod().getAnnotation(Timed.class), invocation.getMethod());
     final Timer.Context timer = metricRegistry.timer(timerTag).time();
-
+    Object result;
     try {
-      return invocation.proceed();
-    } finally {
+      result = invocation.proceed();
+      if (result instanceof Promise<?>) {
+        ((Promise<?>) result).result(v -> timer.stop());
```

Comment:
This isn't right, it will cause the promise to be subscribed to twice.

Would you mind rebasing and using the [new time() method that I just added to Promise](https://github.com/ratpack/ratpack/commit/ea3b351e5046a077481a9a0dffdce95aac4758ef#diff-d0a7bffcb1db1f94820642af231dbaeeR1120)?